core/src/main/java/org/apache/iceberg/rest/auth/HttpRequestFacade.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.auth;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A facade for an HTTP request that allows reading and modifying the request. This is used by
+ * {@link AuthManager} implementations, e.g. to set authentication headers on the request.
+ */
+public interface HttpRequestFacade {
```

Comment:
What is the purpose of calling this a `Facade`? It's an interface so it doesn't matter what the original intent was. This is now Iceberg's model for a HTTP request, so it can be simply `HTTPRequest` (or however we generally capitalize HTTP).

Reply:
I initially called it a facade because it was really an almost 1x1 mapping to `org.apache.hc.core5.http.HttpRequest` – the purpose of this interface was just to avoid leaking implementation details.

I will rename as you suggest but after a quick glance to the code base, it seems that `Http` is preferred over `HTTP` – the only notable exception being `org.apache.iceberg.rest.HTTPClient`.